src/request/src/response.js

```diff
@@ -150,18 +183,20 @@ export class KinveyResponse extends Response {
         || name === 'BlobNotFound'
         || name === 'DocumentNotFound') {
       error = new NotFoundError(message, debug, code, kinveyRequestId);
-    } else if (name === 'ParameterValueOutOfRangeError') {
+    } else if (name === 'ParameterValueOutOfRange') {
       error = new ParameterValueOutOfRangeError(message, debug, code, kinveyRequestId);
     } else if (name === 'ServerError') {
       error = new ServerError(message, debug, code, kinveyRequestId);
+    } else if (name === 'StaleRequest') {
+      error = new StaleRequestError(message, debug, code, kinveyRequestId);
+    } else if (name === 'UserAlreadyExists') {
+      error = new UserAlreadyExistsError(message, debug, code, kinveyRequestId);
+    } else if (name === 'WritesToCollectionDisallowed') {
+      error = new WritesToCollectionDisallowedError(message, debug, code, kinveyRequestId);
     } else {
       return super.error;
```

Comment:
It will return an appropriate error based on the status code or default to a `KinveyError`. See https://github.com/Kinvey/js-sdk/blob/137cb29c64a837efca953c8c8af3b48d8105a271/src/request/src/response.js#L82